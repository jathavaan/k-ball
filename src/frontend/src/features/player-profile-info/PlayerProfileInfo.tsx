import { StyledPlayerInfoCard } from "./playerProfileInfo.style.ts";
import { CardContentText, CardHeader, ImageContainer } from "../ui";
import { PlayerProfileInfoProps } from "./playerProfileInfo.types.ts";
import Grid from "@mui/material/Grid2";
import { Typography } from "@mui/material";

export const PlayerProfileInfo = (props: PlayerProfileInfoProps) => {
  return (
    <StyledPlayerInfoCard data-testid="player-profile-info">
      <Grid container spacing={2} padding="1rem">
        <Grid
          size={{ xs: 12, md: 4 }}
          display="flex"
          flexDirection="column"
          alignItems="center"
        >
          <ImageContainer
            src={props.imageUrl}
            alt={`Portrait of ${props.name}`}
          />
          <CardHeader
            headerText={props.name}
            sx={{
              fontWeight: "bold",
              fontSize: "1.5rem",
              textAlign: "center",
            }}
          />
          <Typography
            variant="body1"
            sx={{ textAlign: "center", marginBottom: "0.5rem" }}
          >
            {props.team}
          </Typography>
          <img
            src={props.teamLogo}
            alt={`Logo of ${props.team}`}
            style={{ width: "2.5rem", height: "2.5rem" }}
          />
        </Grid>

        <Grid size={{ xs: 12, md: 8 }} container spacing={2}>
          <Grid size={{ xs: 6 }}>
            <CardContentText title="Position" text={props.position} />
          </Grid>
          <Grid size={{ xs: 6 }}>
            <CardContentText title="Age" text={props.age.toString()} />
          </Grid>
          <Grid size={{ xs: 6 }} display="flex">
            <CardContentText title="Nationality" text={props.nationality} />
            <img
              src={props.flag}
              alt={`Flag of ${props.nationality}`}
              style={{
                width: "1.25rem",
                height: "1.25rem",
                marginLeft: "0.5rem",
              }}
            />
          </Grid>
          <Grid size={{ xs: 6 }}>
            <CardContentText title="Height" text={`${props.height} cm`} />
          </Grid>
          <Grid size={{ xs: 6 }}>
            <CardContentText title="Place" text={props.place} />
          </Grid>
          <Grid size={{ xs: 6 }}>
            <CardContentText title="Birth Date" text={props.birthDate} />
          </Grid>
          <Grid size={{ xs: 6 }}>
            <CardContentText title="Weight" text={`${props.weight} kg`} />
          </Grid>
        </Grid>
      </Grid>
    </StyledPlayerInfoCard>
  );
};
