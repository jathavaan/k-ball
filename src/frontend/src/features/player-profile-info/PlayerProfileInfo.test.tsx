import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { describe, it, expect, beforeEach } from "vitest";
import { PlayerProfileInfo } from "./PlayerProfileInfo";

describe("PlayerProfileInfo", () => {
  const mockProps = {
    playerId: 2898,
    name: "Kim Young-Gwon",
    team: "Ulsan Hyundai FC",
    teamLogo: "https://media.api-sports.io/football/teams/2767.png",
    imageUrl: "https://media.api-sports.io/football/players/2898.png",
    position: "Defender",
    nationality: "Korea Republic",
    flag: "https://media.api-sports.io/flags/kr.svg",
    age: 34,
    birthDate: "1990-02-27",
    height: 186,
    weight: 74,
    place: "Jeonju, Korea Republic",
  };

  beforeEach(() => {
    render(<PlayerProfileInfo {...mockProps} />);
  });

  it("should render the player's name", () => {
    expect(screen.getByText(mockProps.name)).toBeInTheDocument();
  });

  it("should render the player's age", () => {
    expect(screen.getByText(mockProps.age.toString())).toBeInTheDocument();
  });

  it("should render the player's position", () => {
    expect(screen.getByText(mockProps.position)).toBeInTheDocument();
  });

  it("should render the player's team", () => {
    expect(screen.getByText(mockProps.team)).toBeInTheDocument();
  });

  it("should render the player's nationality", () => {
    expect(screen.getByText(mockProps.nationality)).toBeInTheDocument();
  });

  it("should render the player's place of birth", () => {
    expect(screen.getByText(mockProps.place)).toBeInTheDocument();
  });

  it("should render the player's height", () => {
    expect(screen.getByText(`${mockProps.height} cm`)).toBeInTheDocument();
  });

  it("should render the player's weight", () => {
    expect(screen.getByText(`${mockProps.weight} kg`)).toBeInTheDocument();
  });

  it("should render the player's birth date", () => {
    expect(screen.getByText(mockProps.birthDate)).toBeInTheDocument();
  });
  
  it("should match the snapshot", () => {
    const { asFragment } = render(<PlayerProfileInfo {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
