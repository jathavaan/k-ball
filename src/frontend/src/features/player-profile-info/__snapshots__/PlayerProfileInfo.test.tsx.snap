// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayerProfileInfo > should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class=MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-dynamic-MuiPaper-root-MuiCard-root
    data-testid="player-profile-info"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class=MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-dynamic-MuiGrid2-root
    >
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-dynamic-MuiGrid2-root
      >
        <img
          alt="Portrait of Kim Young-Gwon"
          class="css-1sws5au"
          src="https://media.api-sports.io/football/players/2898.png"
        />
      </div>
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-dynamic-MuiGrid2-root
      >
        <h4
          class=MuiTypography-root MuiTypography-h4 css-dynamic-MuiTypography-root
        >
          Kim Young-Gwon
        </h4>
      </div>
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-dynamic-MuiGrid2-root
      >
        <div
          class="MuiBox-root css-wkpw2c"
        >
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Ulsan Hyundai FC
          </p>
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Current club
          </p>
        </div>
      </div>
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-dynamic-MuiGrid2-root
      >
        <div
          class="MuiBox-root css-wkpw2c"
        >
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Defender
          </p>
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Position
          </p>
        </div>
      </div>
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-dynamic-MuiGrid2-root
      >
        <div
          class="MuiBox-root css-wkpw2c"
        >
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Korea Republic
          </p>
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Nationality
          </p>
        </div>
      </div>
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-dynamic-MuiGrid2-root
      >
        <div
          class="MuiBox-root css-wkpw2c"
        >
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Jeonju, Korea Republic
          </p>
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Birth place
          </p>
        </div>
      </div>
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-dynamic-MuiGrid2-root
      >
        <div
          class="MuiBox-root css-wkpw2c"
        >
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            1990-02-27
          </p>
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Date of birth
          </p>
        </div>
      </div>
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-dynamic-MuiGrid2-root
      >
        <div
          class="MuiBox-root css-wkpw2c"
        >
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            34 years
          </p>
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Age
          </p>
        </div>
      </div>
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-dynamic-MuiGrid2-root
      >
        <div
          class="MuiBox-root css-wkpw2c"
        >
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            186 cm
          </p>
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Height
          </p>
        </div>
      </div>
      <div
        class=MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-dynamic-MuiGrid2-root
      >
        <div
          class="MuiBox-root css-wkpw2c"
        >
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            74 kg
          </p>
          <p
            class=MuiTypography-root MuiTypography-body1 css-dynamic-MuiTypography-root
          >
            Weight
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
